In [3]:
import pandas as pd
import numpy as np

In [4]:
pd.set_option('display.max_columns',None) #display all columns using set option function in data frame 

In [5]:
##Loading the dataset

df =pd.read_csv("../data/raw/train.csv")

In [6]:
##Top rows with head function
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales
0,1,CA-2017-152156,08/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600
1,2,CA-2017-152156,08/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400
2,3,CA-2017-138688,12/06/2017,16/06/2017,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200
3,4,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775
4,5,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680


## Initial Observations from df.head()

1. Dataset contains retail sales transactions.
2. Each row represents a product sold.
3. One order can contain multiple products.
4. Customer information is available.
5. Geographic information is available.
6. Product hierarchy exists (Category → Sub-Category → Product).
7. Sales is the primary business metric.
8. Order Date and Ship Date can be used for time-based analysis.
9. Multiple shipping modes exist.
10. Date columns may require datatype conversion.

In [7]:
##Shape of the dat set
df.shape

(9800, 18)

Observations from shape 
1.  rows : 9800 (transactions)
2.  columns : 18 (fields)

Dataset contains 9,800 sales transactions across 18 business attributes.

In [8]:
##Check Columns names
df.columns.tolist()

['Row ID',
 'Order ID',
 'Order Date',
 'Ship Date',
 'Ship Mode',
 'Customer ID',
 'Customer Name',
 'Segment',
 'Country',
 'City',
 'State',
 'Postal Code',
 'Region',
 'Product ID',
 'Category',
 'Sub-Category',
 'Product Name',
 'Sales']

Observation from Columns name's 

Customer-related columns = 3
Location-related columns = 5
Product-related columns = 4
Sales metric columns = 1
Date columns = 2

In [9]:
##Check data types 
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9800 entries, 0 to 9799
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9800 non-null   int64  
 1   Order ID       9800 non-null   str    
 2   Order Date     9800 non-null   str    
 3   Ship Date      9800 non-null   str    
 4   Ship Mode      9800 non-null   str    
 5   Customer ID    9800 non-null   str    
 6   Customer Name  9800 non-null   str    
 7   Segment        9800 non-null   str    
 8   Country        9800 non-null   str    
 9   City           9800 non-null   str    
 10  State          9800 non-null   str    
 11  Postal Code    9789 non-null   float64
 12  Region         9800 non-null   str    
 13  Product ID     9800 non-null   str    
 14  Category       9800 non-null   str    
 15  Sub-Category   9800 non-null   str    
 16  Product Name   9800 non-null   str    
 17  Sales          9800 non-null   float64
dtypes: float64(2), int6

## Observations from df.info()

1. The dataset contains 9,800 records and 18 columns.
2. Most columns are categorical/string fields.
3. Sales is stored as float64 and is ready for analysis.
4. Postal Code contains 11 missing values.
5. Order Date and Ship Date are stored as strings and require datetime conversion.
6. The dataset includes order, customer, product, and geographical information.
7. Data quality appears good with minimal missing values.
8. The dataset is suitable for sales, customer, product, and regional analysis.

In [10]:
#Missing values 
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
dtype: int64

In [11]:
# Percentage of missing values
round(df.isnull().sum()/len(df)*100,2)

Row ID           0.00
Order ID         0.00
Order Date       0.00
Ship Date        0.00
Ship Mode        0.00
Customer ID      0.00
Customer Name    0.00
Segment          0.00
Country          0.00
City             0.00
State            0.00
Postal Code      0.11
Region           0.00
Product ID       0.00
Category         0.00
Sub-Category     0.00
Product Name     0.00
Sales            0.00
dtype: float64

## Missing Values Analysis

- Postal Code contains 11 missing values (0.11%).
- All other columns have complete data.
- Missing values are minimal and can be handled during data cleaning.

In [12]:
#Duplicate Records

df.duplicated().sum()

np.int64(0)

## Duplicate Records Analysis

- The dataset contains 0 duplicate records.
- Each transaction appears to be unique.
- No duplicate rows need to be removed during the data cleaning phase.
- Data quality is good with respect to duplicate entries.

## EDA (Exploratory Data Analysis)

In [13]:
#describe the data set using describe function

df.describe()

,Row ID,Postal Code,Sales
count,9800.000000,9789.000000,9800.000000
mean,4900.500000,55273.322403,230.769059
std,2829.160653,32041.223413,626.651875
min,1.000000,1040.000000,0.444000
25%,2450.750000,23223.000000,17.248000
50%,4900.500000,58103.000000,54.490000
75%,7350.250000,90008.000000,210.605000
max,9800.000000,99301.000000,22638.480000


## Statistical Summary Observations

1. The dataset contains 9,800 sales transactions.
2. Average sales per transaction is $230.77.
3. Median sales value is $54.49.
4. Since the mean is significantly higher than the median, the sales distribution appears positively skewed.
5. The minimum sales value is $0.44, indicating very small transactions.
6. The maximum sales value is $22,638.48, suggesting the presence of high-value transactions or potential outliers.
7. 75% of transactions are below $210.61, indicating that most sales are relatively small.
8. A small number of large transactions contribute significantly to overall revenue.